src/features/callAssignments/components/CallAssignmentStatusCards.tsx

```diff
@@ -32,6 +50,81 @@ const CallAssignmentStatusCards = ({
           />
           <List>
             <StatusCardItem
+              action={
+                <Box display="flex" justifyContent="space-between">
+                  <Typography color="secondary" variant="h5">
+                    <Msg
+                      id="pages.organizeCallAssignment.blocked.hours"
+                      values={{ cooldown }}
+                    />
+                  </Typography>
+                  <Box ml={1}>
+                    <SettingsIcon
+                      color="secondary"
+                      cursor="pointer"
+                      onClick={(event) =>
+                        setAnchorEl(anchorEl ? null : event.currentTarget)
+                      }
+                    />
+                  </Box>
+                  <Popper anchorEl={anchorEl} open={!!anchorEl}>
+                    <ClickAwayListener
+                      onClickAway={() => {
+                        setAnchorEl(null);
+                        if (newCooldown != null && newCooldown != cooldown) {
+                          model.setCooldown(newCooldown);
+                        }
+                      }}
+                    >
+                      <Paper elevation={3} variant="elevation">
+                        <Box mt={1} p={2}>
+                          <TextField
+                            helperText={intl.formatMessage({
+                              id: 'pages.organizeCallAssignment.blocked.cooldownHelperText',
+                            })}
+                            label={intl.formatMessage({
+                              id: 'pages.organizeCallAssignment.blocked.cooldownLabel',
+                            })}
+                            onChange={(ev) => {
+                              const val = ev.target.value;
+
+                              if (val == '') {
+                                setNewCooldown(null);
+                                return;
+                              }
+
+                              const intVal = parseInt(val);
+                              if (!isNaN(intVal) && intVal.toString() == val) {
+                                setNewCooldown(intVal);
+                              }
+                            }}
+                            onKeyDown={(ev) => {
+                              if (ev.key === 'Enter') {
+                                if (
+                                  newCooldown != null &&
+                                  newCooldown != cooldown
+                                ) {
```

Comment:
Maybe this type of validation should be part of the model somehow?

Reply:
i put this as "extra security", there is already a check in the component that only runs the method if the new cooldown is different from the old one. Is it usually so that you check in both places, or can be removed in one place?